cmd/check/main.go

```diff
@@ -166,16 +171,38 @@ func checkRepository(principal resource.BasicCredentials, version *resource.Vers
 	digestVersions := map[string]*semver.Version{}
 	for _, identifier := range tags {
 		var ver *semver.Version
-		if identifier == "latest" {
-			// TODO: handle variants, e.g. 'alpine' is latest for
-			// 'x.x.x-alpine'
+		if identifier == bareTag {
 			latestTag = identifier
 		} else {
-			ver, err = semver.NewVersion(identifier)
+			verStr := identifier
+			if variant != "" {
+				if !strings.HasSuffix(identifier, "-"+variant) {
+					continue
+				}
+
+				verStr = strings.TrimSuffix(identifier, "-"+variant)
+			}
+
+			ver, err = semver.NewVersion(verStr)
 			if err != nil {
 				// not a version
 				continue
 			}
+
+			pre := ver.Prerelease()
+			if pre != "" {
+				// contains additional variant
+				if strings.Contains(pre, "-") {
+					continue
+				}
+
+				if !strings.HasPrefix(pre, "alpha.") &&
+					!strings.HasPrefix(pre, "beta.") &&
+					!strings.HasPrefix(pre, "rc.") {
```

Comment:
What about tags that don't follow this format exactly? e.g. https://hub.docker.com/r/kubernetesui/dashboard/tags?page=1&name=v2.0.0-rc7, doesn't have a `.` after `rc`? This may not be a valid semver tag, not sure - but it looks like `1.2.3-alpha` is, but wouldn't be supported

Reply:
good point - i think that's easy to support. pushed a fix!